swupd/packs_test.go

```diff
@@ -537,3 +537,106 @@ func TestTwoDeltasForTheSameTarget(t *testing.T) {
 		ts.checkExists(fmt.Sprintf("www/20/delta/10-20-%s-%s", hashA, hashB))
 	}
 }
+
+func TestPackRenames(t *testing.T) {
+	ts := newTestSwupd(t, "packing-renames-")
+	defer ts.cleanup()
+
+	ts.Bundles = []string{"os-core"}
+
+	content := strings.Repeat("CONTENT", 1000)
+
+	// Version 10.
+	ts.write("image/10/os-core/file1", content+"10")
+	ts.createManifests(10)
+
+	// Version 20 changes file1 contents.
+	ts.write("image/20/os-core/file1", content+"20")
+	ts.createManifests(20)
+
+	// Version 30 changes file1 and renames to file2.
+	ts.write("image/30/os-core/file2", content+"30")
+	ts.createManifests(30)
+
+	m := ts.parseManifest(30, "os-core")
+	file1 := fileInManifest(t, m, 30, "/file1")
+	file2 := fileInManifest(t, m, 30, "/file2")
+	checkRenamePair(t, file1, file2)
+
+	hashIn10 := ts.mustHashFile("image/10/os-core/file1")
+	hashIn20 := ts.mustHashFile("image/20/os-core/file1")
+	hashIn30 := ts.mustHashFile("image/30/os-core/file2")
+
+	// Pack from 10->20 will contain a delta due to content change.
+	info := ts.createPack("os-core", 10, 20, ts.path("image"))
+	mustHaveDeltaCount(t, info, 1)
+	checkFileInPack(t, ts.path("www/20/pack-os-core-from-10.tar"), fmt.Sprintf("delta/10-20-%s-%s", hashIn10, hashIn20))
+
+	// Pack from 20->30 will contain a delta due to content change (and rename).
+	info = ts.createPack("os-core", 20, 30, ts.path("image"))
+	mustHaveDeltaCount(t, info, 1)
+	checkFileInPack(t, ts.path("www/30/pack-os-core-from-20.tar"), fmt.Sprintf("delta/20-30-%s-%s", hashIn20, hashIn30))
+
+	// Pack from 10->30 will contain a delta due to content change (and rename).
+	info = ts.createPack("os-core", 10, 30, ts.path("image"))
+	mustHaveDeltaCount(t, info, 1)
+	checkFileInPack(t, ts.path("www/30/pack-os-core-from-10.tar"), fmt.Sprintf("delta/10-30-%s-%s", hashIn10, hashIn30))
+
+	// TODO: When/If rename sticks flag, we can add a Version 40, that change an
```

Comment:
*sigh* you know my thoughts on this. Have a fix in #154 though.

Reply:
I'll rebase when it lands. The "/If" part was more because at the time I asked there was no consensus about this.

Reply:
rename persistence now fixed and merged.

Reply:
Wrote the test and removed the comment.

(Now I figured that maybe your concern was about having a TODO instead of a failing test. We are lucky this is a non-problem again :-)